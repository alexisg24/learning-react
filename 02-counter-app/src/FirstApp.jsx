import PropTypes from 'prop-types'

const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid='test-title'> {title} </h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  )
}

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
  title: 'Your Hello World Here',
  subTitle: 'Your SubTitle Here!',
  name: 'Your Name Here!'
}

export { FirstApp }
