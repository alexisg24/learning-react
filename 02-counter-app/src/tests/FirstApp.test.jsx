import { render } from '@testing-library/react'
import { FirstApp } from '../FirstApp'

describe('Tests in <FirstApp />', () => {
  test('Needs to match with the snapshot', () => {
    const title = 'Hola, soy Goku'
    const { container } = render(<FirstApp title={title} />)
    expect(container).toMatchSnapshot()
  })

  test('Needs to show title in h1', () => {
    const title = 'Hola, soy Goku'
    const { container, getByText } = render(<FirstApp title={title} />)
    expect(getByText(title)).toBeTruthy()

    const h1 = container.querySelector('h1')
    expect(h1.innerHTML).toContain(title)
  })
})
