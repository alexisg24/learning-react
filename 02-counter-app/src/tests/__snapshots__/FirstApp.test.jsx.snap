// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> Needs to match with the snapshot 1`] = `
<div>
  <h1>
     
    Hola, soy Goku
     
  </h1>
  <p>
    Your SubTitle Here!
  </p>
  <p>
    Your Name Here!
  </p>
</div>
`;
